tutorials/packaging/continuous-snap-delivery-from-circle-ci/continuous-snap-delivery-from-circle-ci.md

```diff
@@ -0,0 +1,497 @@
+
+If you already have your project in GitHub packaged as a snap, you can skip this step and jump straight to the next one. If you have your project in GitHub but haven’t packaged it yet, we have many tutorials that can help! The "[Create your first snap]" tutorial is a good place to start.
+
+Let's set up one GitHub project here, in case you don't already have one.
+
+First, go to [https://github.com] and sign up. Then, go to [https://github.com/new] where we will fill the details of our new repository. The only required field is the repository name where you can enter something like `hello-snap-yourusername`. That might look something like this:
+
+![](images/new_repository.png)
+
+Next, click the "Create repository" button to get an empty repository in GitHub. Let's put our snap project in there!
+
+Open a terminal, and run the following commands. The first one is to install git:
+
+``` bash
+sudo apt install git
+```
+
+Now, clone your repository to your local workspace, using the address displayed in GitHub. Something like this, but with your username:
+
+
+``` bash
+mkdir -p ~/workspace
+cd workspace
+git clone git@github.com:kyrofa/hello-snap-username.git
+cd hello-snap-username
+```
+
+To create a snap project, open a text editor, and write the following content:
+
+``` yaml
+name: hello-yourname
+version: '2.10'
+summary: GNU Hello, the "hello world" snap
+description: GNU hello prints a friendly greeting.
+grade: stable
+confinement: strict
+
+apps:
+  hello:
+    command: bin/hello
+
+parts:
+  gnu-hello:
+    source: http://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
+    plugin: autotools
+```
+
+Save it to `~/workspace/hello-snap-username/snap/snapcraft.yaml` (remember to replace "username" with yours!)
+
+This is the packaging metadata of a very simple snap, that, as the description says, will just print a greeting. The "[Create your first snap]" tutorial can provide details for all the lines of this file, so please refer to it to know more about the packaging. In here we will be focusing just on the continuous delivery, so let's continue and push this snap to our GitHub repository:
+
+
+``` bash
+git add snap/snapcraft.yaml
+git commit -m "add the hello snap"
+git push
+```
+
+Now that we have chosen our snap name, we need to make sure we register it to our store account (snap names are unique, and they must be registered before they can be published). Replace "hello-yourname" with whatever you used for the `name` field in the `snapcraft.yaml`.
+
+```bash
+snapcraft login
+snapcraft register hello-yourname
+```
+
+We now have a GitHub project containing packaging details for a snap that has been registered to our account. Let's move onto Circle CI.
+
+## Setting up Circle CI
+Duration: 5:00
+
+To use Circle CI, first we will have to sign up there. Let's go to [https://circleci.com/vcs-authorize] and click "Log In with GitHub". Then, click "Authorize application" to allow Circle CI to access your GitHub account.
```

Comment:
Link formatting needs fixing

Reply:
Dang, I knew I was gonna miss at least one! Thank you-- fixed.